src/Microsoft.AspNetCore.Mvc.RazorPages/DependencyInjection/RazorPagesOptionsExtensions.cs

```diff
@@ -170,6 +170,50 @@ public static RazorPagesOptions AuthorizeFolder(this RazorPagesOptions options,
         public static RazorPagesOptions AuthorizeFolder(this RazorPagesOptions options, string folderPath) =>
             AuthorizeFolder(options, folderPath, policy: string.Empty);
 
+        /// <summary>
+        /// Associates a friendly rotue with the page at the specified <paramref name="path"/>.
+        /// </summary>
+        /// <param name="options">The <see cref="RazorPagesOptions"/>.</param>
+        /// <param name="path">The page path.</param>
+        /// <param name="routeUrl">The route to associate with the page.</param>
+        /// <returns>The <see cref="RazorPagesOptions"/>.</returns>
+        public static RazorPagesOptions MapPageRoute(this RazorPagesOptions options, string path, string routeUrl)
```

Comment:
We should revisit this again in p3, we don't in general want to put extension methods directly on options